client/controller/shared.js

```diff
@@ -35,6 +37,62 @@ export function makeLayoutMiddleware( LayoutComponent ) {
 	};
 }
 
+/***
+ * Replaces stylesheet element gracefully
+ *
+ * If we set the `href` attribute of `link` directly, the browser will remove
+ * the previous stylesheet and will start loading the new one, but in the meantime
+ * we won't have all of the sizes the old one had.
+ *
+ * To prevent that, we first load a new stylesheet and then remove the old one
+ *
+ * @param {DOMElement} linkElement Link to change
+ * @param {String} newHref new url to load
+ */
+function replaceStlyesheet( linkElement, newHref ) {
+	const newLink = Object.assign( document.createElement( 'link' ), {
+		rel: 'stylesheet',
+		type: 'text/css',
+		href: newHref,
+	} );
+
+	if ( linkElement.id ) {
+		newLink.id = linkElement.id;
```

Comment:
We shouldn't have 2 elements with the same id at the same time, we should assign the id once we remove the old element imo